neon_llm_core/chatbot.py

```diff
@@ -73,84 +78,115 @@ def ask_discusser(self, options: dict, context: dict = None) -> str:
         """
         queue = self.mq_queue_config.ask_discusser_queue
-        return send_mq_request(vhost=self.mq_queue_config.vhost,
-                               request_data={"query": prompt,
-                                             "options": options,
-                                             "persona": self.persona},
-                               target_queue=queue,
-                               response_queue=f"{queue}.response")
-
-    def _get_llm_api_choice(self, prompt: str, responses: List[str]) -> dict:
+        try:
+            request_data = LLMDiscussRequest(model=self.base_llm,
+                                             persona=self.persona,
+                                             query=prompt,
+                                             options=options,
+                                             history=[],
+                                             message_id="")
+            resp_data = send_mq_request(vhost=self.mq_queue_config.vhost,
+                                        request_data=request_data.model_dump(),
+                                        target_queue=queue,
+                                        response_queue=f"{queue}.response")
+            return LLMDiscussResponse(**resp_data)
+        except Exception as e:
+            LOG.exception(f"Failed to get response on "
+                          f"{self.mq_queue_config.vhost}/{queue}: {e}")
+
+    def _get_llm_api_choice(self, prompt: str,
+                            responses: List[str]) -> Optional[LLMVoteResponse]:
         """
         Requests LLM API for choice among provided message list
         :param prompt: incoming prompt text
         :param responses: list of answers to select from
         :returns response data from LLM API
         """
         queue = self.mq_queue_config.ask_appraiser_queue
-        return send_mq_request(vhost=self.mq_queue_config.vhost,
-                               request_data={"query": prompt,
-                                             "responses": responses,
-                                             "persona": self.persona},
-                               target_queue=queue,
-                               response_queue=f"{queue}.response")
+
+        try:
+            request_data = LLMVoteRequest(model=self.base_llm,
+                                          persona=self.persona,
+                                          query=prompt,
+                                          responses=responses,
+                                          history=[],
+                                          message_id="")
+            resp_data = send_mq_request(vhost=self.mq_queue_config.vhost,
+                                        request_data=request_data.model_dump(),
+                                        target_queue=queue,
+                                        response_queue=f"{queue}.response")
+            return LLMVoteResponse(**resp_data)
```

Comment:
I would omit kwargs-like argument passing here, to make it more flexible to any potential updates to the callback body of send_mq_request